frontend-v2/src/app/interest/generator/generator-form.tsx

```diff
@@ -0,0 +1,264 @@
+                <FormControl>
+                  <Checkbox
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                  />
+                </FormControl>
+                <FormLabel>
+                  Show "What are your activism interests, if any?"
+                </FormLabel>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="referralApplyName"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Pre-fill "Who encouraged you to sign up?"</FormLabel>
+                <FormControl>
+                  <Input type="text" placeholder="Referral" {...field} />
+                </FormControl>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="referralFriends"
+            render={({ field }) => (
+              <FormItem className="flex flex-row items-center gap-2 space-y-0">
+                <FormControl>
+                  <Checkbox
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                  />
+                </FormControl>
+                <FormLabel>
+                  Show "List any existing DxE activists who you are close
+                  friends with"
+                </FormLabel>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="referralApply"
+            render={({ field }) => (
+              <FormItem className="flex flex-row items-center gap-2 space-y-0">
+                <FormControl>
+                  <Checkbox
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                  />
+                </FormControl>
+                <FormLabel>Show "Who encouraged you to sign up?"</FormLabel>
+              </FormItem>
+            )}
+          />
+
+          <FormField
+            control={form.control}
+            name="referralOutlet"
+            render={({ field }) => (
+              <FormItem className="flex flex-row items-center gap-2 space-y-0">
+                <FormControl>
+                  <Checkbox
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                  />
+                </FormControl>
+                <FormLabel>
+                  Show "Where did you hear about this opportunity to get
+                  involved in DxE?"
+                </FormLabel>
+              </FormItem>
+            )}
+          />
+
+          <div>
+            <Button type="submit" className="w-full">
+              Generate URL
+            </Button>
+          </div>
+
+          <FormItem>
+            <FormLabel>URL</FormLabel>
+            <FormControl>
+              <Textarea readOnly value={output} />
```

Comment:
✨ might be nice to have a "copy" button by this

Reply:
done